GAD2.py

```diff
@@ -0,0 +1,209 @@
+import os
+import re
+from typing import TextIO, Optional, Union
+from abc import ABC, abstractmethod
+from Bio import SeqIO, SeqUtils
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__():
+        pass
+
+    def __getitem__():
+        pass
+
+    def __str__():
+        pass
+
+    def check_sequence():
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    """Proccising nucleic acid sequences.
+    This class is parental for DNASequence and RNASequence classes.
+    """
+    def __init__(self, seq) -> None:
+        self.seq = seq
+
+    def check_sequence(self):
+        """
+        Checkes the correctnessis of input string (DNA or RNA).
+        """
+        for nucleotide in self.seq:
+            if not nucleotide in type(self).complement_rule:
+                raise ValueError (f'Incorrect nucleotide in {self.seq}')
+        return True
+
+    def complement(self):
+        """
+        Create complement sequences of RNA or DNA acoording to the complementary base pairing rule. 
+        """
+        if self.check_sequence():
+            complement_sequence = ''
+            for nucleotide in self.seq:
+                if nucleotide in type(self).complement_rule:
+                    complement_sequence += type(self).complement_rule[nucleotide]
+            return type(self)(complement_sequence)
+
+    def gc_content(self):
+        return (sum(1 for _ in re.finditer(r'[GCgc]', self.seq)))/self.__len__()
+    
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, key):
+        return self.seq[key]
+
+    def __str__(self):
+        return self.seq
+
+class DNASequence(NucleicAcidSequence):
+    """
+    Procissing DNA sequences.
+    Argumemt: seq (str) - DNA sequence, letters case do not matter.
+    -Example nucl_seq = DNASequence('ATGCGC' OR 'ATgCgc' OR 'atgcgc).
+
+    Valid operations:
+    - transcribe - return transcibed sequence of DNA coding strand;
+    - complement - return complementary sequence according to complementary rule
+    - check sequence - is input sequence correct
+    - gc_content - return percent of GC in sequence
+    """
+    complement_rule = {'a': 't', 'A': 'T', 't': 'a', 'T': 'A',
+                       'g': 'c', 'G': 'C', 'c': 'g', 'C': 'G'}
+    
+    def __init__(self, seq: str) -> None:
+        super().__init__(seq = seq)
+
+    def transcribe(self):
+        """
+        Return transcribed sequence of DNA acoording to the complementary base pairing rule.  
+        Function can procced only DNA seqences.
+        """
+        transcribed_sequence = ''
+        transcribed_sequence = self.seq.replace('t', 'u').replace('T', 'U')
+        return type(self)(transcribed_sequence)
```

Comment:
Вот тут можно сделать. Тогда транскрибированная ДНК превратится в РНК, и будет вообще круто.
```suggestion
        return RNASequence(transcribed_sequence)
```